armi/nuclearDataIO/xsCollections.py

```diff
@@ -785,13 +784,44 @@ def computeMacroscopicGroupConstants(
     multConstant=None,
     multLib=None,
 ):
-    """
+    r"""
     Compute any macroscopic group constants given number densities and a microscopic library.
 
     .. impl:: Compute macroscopic cross sections from microscopic cross sections and number densities.
         :id: I_ARMI_NUCDATA_MACRO
         :implements: R_ARMI_NUCDATA_MACRO
 
+        This method computes the macroscopic cross sections of a specified
+        reaction type from inputted microscopic cross sections and number
+        densities. The ``constantName`` parameter specifies what type of
+        reaction is requested. The ``numberDensities`` parameter is dictionary
+        mapping the nuclide to its density. The ``lib`` parameter is a library
+        object like :py:class:`~armi.nuclearDataIO.xsLibraries.IsotxsLibrary` or
+        :py:class:`~armi.nuclearDataIO.xsLibraries.CompxsLibrary` that holds the
+        microscopic cross-section data. The ``microSuffix`` parameter specifies
+        from which part of the library the microscopic cross sections are
+        gathered; this is typically gathered from a components
+        ``getMicroSuffix`` method like :py:meth:`Block.getMicroSuffix
+        <armi.reactor.blocks.Block.getMicroSuffix>`. ``libType`` is an optional
+        parameter specifying whether the reaction is for neutrons or gammas.
+        This method also has the optional parameters ``multConstant`` and
+        ``multLib``, which allows another reaction's cross section to be
+        multiplied to the primary one. The macroscopic cross are then compute
+        as:
+
+        .. math::
+
+            \Sigma_{g} = \sum_{n} N_n \sigma_{n,g} \sigma_{\mathrm{m}, n,
+            g} \quad g=1,...,G
+
+        where :math:`n` is the isotope index, :math:`g` is the energy group
+        index, :math:`\sigma` is the microscopic cross section, and
+        :math:`\sigma_{\mathrm{m}}` is the multiplying microscopic cross
+        section. If the ``constantName`` reaction is missing a cross section for
+        one or more of the nuclides in ``numberDensities`` a error is raised;
```

Comment:
I don't think this is right. An error is thrown if a given nuclide + suffix pair can't be found in the library object that's passed in (it gets added to `skippedNuclide`).

Reply:
Looks like an error to me:
```python
    if skippedNuclides:
        msg = "The following nuclides are not in microscopic library {}: {}".format(
            lib, skippedNuclides
        )
        runLog.error(msg, single=True)
        raise ValueError(msg)
```

Reply:
Sorry, didn't ready your comment carefully. I'll add the caveat that it must be in the `lib` with the given `microSuffix`

Reply:
Cool. I think it makes sense now.